architecture/adr-014-module-catalog.md

```diff
@@ -0,0 +1,21 @@
+<!-- Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License"). You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0. See the NOTICE file distributed with this work for additional information regarding copyright ownership. Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License. -->
+
+| Type          | ID            | Status        | Title                  |
+| ------------- | ------------- | ------------- | ---------------------- |
+| ADR           | adr-014       | Proposed      | Multiple ModuleCatalog |
+
+
+# Architecture Decision Record: Multiple ModuleCatalog
+
+## Context
+ComposeUI aims to support many different kind of modules and applications. These applications/modules, that can be started via the ModuleLoader should be stored in ModuleCatalogs and every module should have it's unique module id. The user should have the ability to handle multiple ModuleCatalog with the ModuleLoader.
+
+General problem statement:
+
+- When multiple ModuleCatalog is being used, some modules could have the same module id in different ModuleCatalog which could cause issue while starting/stopping the application/module via the ModuleLoader.
+
+## Decision
```

Comment:
Just an idea: what if we'd fail fast & hard if there are duplicates, i.e.: throw an exception and crash? What do you think?

Reply:
To recap, the issue with failing on duplicates from different sources is that it might cause a situation that the developer/user has no way of resolving.
Ideally this is never an issue but if merging multiple module catalogs is a feature we want to support and we wish to keep the standard interface (and not do something like e.g. prefixing ids that would be non-standard), we need a well documented consistent way of resolving these conflicts that the developer/user has a way to control.